### PR TITLE
Add a check to make sure yarn.lock was properly updated

### PR DESCRIPTION
**Pull request by rsimha-amp**
This PR adds a new check to safeguard against merge conflicts in `yarn.lock` like the one in https://github.com/ampproject/amphtml/pull/13067#issuecomment-362022629.

It also fixes the out-of-sync `yarn.lock` on `master`.

### CODE REVIEW

**Comment by rsimha-amp** [comment]
Use this link to filter out whitespace-only changes during review: https://github.com/ampproject/amphtml/pull/13183/files?w=1

**Comment by rsimha-amp** [comment]
Example of this check in action: https://travis-ci.org/ampproject/amphtml/jobs/335772096

This PR is now ready for a full review.

File: build-system/pr-check.js [hunk 1]
```diff
@@ -453,24 +453,36 @@ function main() {
...
-          colors.cyan('yarn.lock'));
-      console.error(fileLogPrefix, colors.yellow('NOTE:'),
-          'To update', colors.cyan('yarn.lock'), 'after changing',
-          colors.cyan('package.json') + ',', 'run',
-          '"' + colors.cyan('yarn install') + '"',
-          'and include the updated', colors.cyan('yarn.lock'),
-          'in your PR.');
-      process.exit(1);
-    }
+  const yarnIntegrityCheck = getStderr('yarn check --integrity').trim();
```
Comment (cathyxz) [inline build-system/pr-check.js]: So legit.

File: build-system/pr-check.js [hunk 2]
```diff
@@ -389,6 +389,47 @@ function runAllCommandsLocally() {
...
+        'in your PR.');
+    process.exit(1);
+  }
+}
+
+/**
+ * Makes sure that yarn.lock was properly updated.
+ */
+function runYarnLockfileCheck() {
+  const yarnLockfileCheck = getStdout('git -c color.ui=always diff').trim();
```
Comment (choumx) [inline build-system/pr-check.js]: This doesn't seem right.

Comment (rsimha-amp) [reply build-system/pr-check.js]: What about it doesn't seem right?

The reason we check using `git diff` is that the result should be empty on Travis during a normal PR check. If any uncommitted diffs show up on Travis, they're going to be in `yarn.lock`, as a result of `yarn` being run.

Comment (choumx) [reply build-system/pr-check.js]: Derp, I missed the `diff` at the end. Nevermind and please proceed.

**Comment by cvializ** [comment]
Thanks Raghu! ☺️ Looks like I borked a rebase conflict.

**Pull request merged**